client/src/app/App.js

```diff
@@ -1436,14 +1436,40 @@ export class App extends PureComponent {
 
     const contents = await this.tabRef.current.triggerAction('save');
 
-    return fileSystem.writeFile(savePath, {
+    const file = await fileSystem.writeFile(savePath, {
       ...originalFile,
       contents
     }, {
       encoding,
       fileType: saveType
     });
 
+    if (originalFile.path !== savePath) {
+      await this.migrateConfigForFilePath(originalFile.path, savePath);
+    }
+
+    return file;
+  }
+
+  /**
+   * Migrate configurations from old file path to new file path.
+   *
+   * @param {string|null} oldPath - Original file path (null for new/unsaved files)
+   * @param {string} newPath - New file path
+   */
+  async migrateConfigForFilePath(oldPath, newPath) {
+    if (!newPath || !oldPath) {
+      return;
+    }
+
+    const config = this.getGlobal('config');
+
+    const fileConfig = await config.getConfigForFile({ path: oldPath });
```

Comment:
Related to https://github.com/camunda/camunda-modeler/pull/5464/files/45573351d17f5f0a5db21bb394c737de34212fc5#r2623412215
```suggestion
    const fileConfig = await config.getForFile({ path: oldPath });
```